backend/kesaseteli/companies/services.py

```diff
@@ -40,5 +53,45 @@ def get_or_create_company_with_name_and_business_id(
     company, _ = Company.objects.get_or_create(
         name=name,
         business_id=business_id,
+        eauth_profile=eauth_profile,
     )
     return company
+
+
+def get_or_create_company_from_eauth_profile(
+    eauth_profile: EAuthorizationProfile,
+) -> Company:
+    company = Company.objects.filter(eauth_profile=eauth_profile).first()
+
+    if not company:
+        if settings.MOCK_FLAG:
+            company = CompanyFactory(eauth_profile=eauth_profile)
+        else:
+            try:
+                organization_roles = get_organization_roles(eauth_profile)
+            except HTTPError:
+                raise NotFound(
+                    detail="Unable to fetch organization roles from eauthorizations API"
+                )
+
+            business_id = organization_roles.get("identifier")
+
+            company = Company.objects.filter(business_id=business_id).first()
+
+            if not company:
```

Comment:
Just an idea - Would it make sense to add a check for missing company.ytj_json and retry to retrieve the data again, if it failed previously? (don't know - if it adds a lot of delay then it would certainly not make sense)

Reply:
Hmm, that would certainly make sense! The delay is not too bad so I'll add it.